### ISSUE TITLE
magictype should be check for None value

### ISSUE THREAD

**Issue by dfroger** [pazz/alot#841]
The [buffer function](https://github.com/threatstack/libmagic/blob/1249b5cd02c3b6fb9b917d16c76bc76c862932b6/python/magic.py#L133) of `libmagic` may return `None`, but when `guess_mimetype` [call the buffer function](https://github.com/pazz/alot/blob/9fb17f2f6a4b15da66d478cb7c030d3f2daa63c8/alot/helper.py#L398), it doesn't check for the `None` value, and [re.match(r'\w+\/\w+', magictype)](https://github.com/pazz/alot/blob/9fb17f2f6a4b15da66d478cb7c030d3f2daa63c8/alot/helper.py#L405) fails (when I try to reply to an email):

```
Traceback (most recent call last):
  File "/XXX/lib/python2.7/site-packages/twisted/internet/defer.py", line 588, in _runCallbacks
    current.result = callback(current.result, *args, **kw) cluster (inspiré de la liste d'issues sur bitbucket ;) ) :
  File "/XXX/lib/python2.7/site-packages/alot/ui.py", line 633, in call_apply
    return defer.maybeDeferred(cmd.apply, self) problèmes, etc...)
  File "/XXX/lib/python2.7/site-packages/twisted/internet/defer.py", line 150, in maybeDeferred
    result = f(*args, **kw)da (*)
  File "/XXX/lib/python2.7/site-packages/twisted/internet/defer.py", line 1274, in unwindGenerator
    return _inlineCallbacks(None, gen, Deferred())
--- <exception caught here> --- de noeuds gpu
  File "/XXX/lib/python2.7/site-packages/twisted/internet/defer.py", line 1128, in _inlineCallbacks
    result = g.send(result)
  File "/XXX/lib/python2.7/site-packages/alot/commands/globals.py", line 782, in apply
    mimetype = helper.guess_mimetype(sigcontent)
  File "/XXX/lib/python2.7/site-packages/alot/helper.py", line 409, in guess_mimetype
    if re.match(r'\w+\/\w+', magictype):
  File "/XXX/lib/python2.7/re.py", line 141, in match
    return _compile(pattern, flags).match(string)
exceptions.TypeError: expected string or buffer
```

**Comment by dfroger** [comment]
I don't have the problem if I use `magic` from `pip`:

     pip install python-magic

**Comment by pazz** [comment]
the problem is that there are ca 1000 competing libmagic's for python and at some point we decided to allow both the variant that's on pip (for the believers) and the one that comes with libmagic and is available in debian packaging. In any case, i suppose this merrits a hotfix.

**Comment by pazz** [comment]
pushed a fix to `0.3.8-magic-841` and `testing` branches. Let me know if problems remain.
